app-modules/project/src/Policies/ProjectPolicy.php

```diff
@@ -68,14 +93,60 @@ public function create(Authenticatable $authenticatable): Response
 
     public function update(Authenticatable $authenticatable, Project $project): Response
     {
+        if (ProjectManagersAuditorsFeature::active() && ! auth()->user()->isSuperAdmin()) {
+            $team = auth()->user()->team;
+
+            /** @var Collection<int, Team> $managerTeams */
+            $managerTeams = $project->managerTeams;
+            /** @var Collection<int, Team> $managerUsers */
+            $managerUsers = $project->managerUsers;
+            /** @var Collection<int, Team> $auditorTeams */
+            $auditorTeams = $project->auditorTeams;
+            /** @var Collection<int, Team> $auditorUsers */
+            $auditorUsers = $project->auditorUsers;
+
+            if (
+                ! $managerTeams->contains('id', $team?->getKey()) &&
+                ! $auditorTeams->contains('id', $team?->getKey()) &&
+                ! $managerUsers->contains('id', auth()->user()->getKey()) &&
+                ! $auditorUsers->contains('id', auth()->user()->getKey()) &&
+                ! $project->createdBy?->is(auth()->user())
+            ) {
+                return Response::deny("You don't have permission to update this project because you're not an auditor or manager or creator of this project.");
+            }
```

Comment:
In order to edit or delete, the user must be a super admin, the creator of the project, or a **manager**. Auditors are not allowed to take these actions. So please remove an details of being an auditor here.

Also, try and see if this can be changed to queries that return a bool with like `exists` rather than having to load users and teams into memory.